docs/articles_en/documentation/openvino-ir-format/operation-sets/operation-specs/internal/moe.rst

```diff
@@ -0,0 +1,151 @@
+
+**Detailed description**:  
+The MOE op receives hidden states, routing weights, and indices of selected experts, along with expert weights and (optionally) biases. It performs the expert computation as specified by the ``expert_type`` attribute, applying the routing_weights and combining the results. This enables efficient, fused computation of Mixture of Experts architectures excluding the router part (computation of routing weights).
+
+**Pseudocode for expert_type**
+The ``router_topk_output_indices`` are used to select the top-k experts for optimized computation, not included in the pseudocode below.
+
+* ``GEMM2_BIAS_SWIGLU_CLAMP``:
+
+.. code-block:: py
+    :force:
+
+    # Common part: Reshape hidden states and prepare for expert computation
+    reshaped_hidden_states = reshape(hidden_states, [-1, 0], special_zero=True)
+    tiled_hidden_states = tile(reshaped_hidden_states, [num_experts, 1])
+    reshaped_hidden_states = reshape(tiled_hidden_states, [num_experts, -1, 0], special_zero=True)
+
+    # Experts computation part (GEMM2_BIAS_SWIGLU_CLAMP)
+    # Fused gate_up computation
+    gate_up = matmul(reshaped_hidden_states, weight_0, transpose_a=False, transpose_b=False) + bias_0
+    # Slice gate_up into two halves along last dimension, taking every second element with step two
+    slice_1 = gate_up[..., ::2]      # every second element starting from index 0
+    slice_2 = gate_up[..., 1::2]     # every second element starting from index 1
+    # Branch 1: Minimum and Swish
+    minimum_1 = minimum(slice_2, expert_alpha)
+    swish_1 = swish(minimum_1, beta=expert_beta)
+    # Branch 2: Clamp and Add
+    clamp_1 = clamp(slice_1, -expert_alpha, expert_alpha)
+    add_1 = clamp_1 + 1
+    # Multiply branches
+    fused = add_1 * swish_1
+    # Down projection
+    down_proj = matmul(fused, weight_1, transpose_a=False, transpose_b=False) + bias_1
+
+    # Common part: Routing and summation
+    routed_experts = reshape(down_proj, [num_experts, batch_size, -1, hidden_size]) * routing_weights
+    output = reduce_sum(routed_experts, axis=0, keep_dims=False)
+
+* ``GEMM3_SWIGLU``:
+
+.. code-block:: py
+    :force:
+
+    # Common part: Reshape hidden states and prepare for expert computation
+    reshaped_hidden_states = reshape(hidden_states, [-1, 0], special_zero=True)
+    tiled_hidden_states = tile(reshaped_hidden_states, [num_experts, 1])
+    reshaped_hidden_states = reshape(tiled_hidden_states, [num_experts, -1, 0], special_zero=True)
+
+    # Experts computation part (GEMM3_SWIGLU)
+    x_proj = matmul(reshaped_hidden_states, weight_0, transpose_a=False, transpose_b=True)
+    x_proj2 = matmul(reshaped_hidden_states, weight_1, transpose_a=False, transpose_b=True)
+    swiglu = swish(x_proj, beta=expert_beta)
+    x_proj = x_proj2 * swiglu
+    down_proj = matmul(x_proj, weight_2, transpose_a=False, transpose_b=True)
```

Comment:
GPU plugin request is to transpose those weights at conversion stage, so the `MatMul` both `transpose_a/b` attrs should be `False` at this point:
```suggestion
    # Experts computation part (GEMM3_SWIGLU)
    x_proj = matmul(reshaped_hidden_states, weight_0, transpose_a=False, transpose_b=False)
    x_proj2 = matmul(reshaped_hidden_states, weight_1, transpose_a=False, transpose_b=False)
    swiglu = swish(x_proj, beta=expert_beta)
    x_proj = x_proj2 * swiglu
    down_proj = matmul(x_proj, weight_2, transpose_a=False, transpose_b=False)
```
cc: @yeonbok